testing/test_deploy.py

```diff
@@ -224,7 +223,7 @@ def teardown(args):
   core_api.delete_namespace(args.namespace, {})
 
 def determine_test_name(args):
-  return args.func.__name__
+  return args.func.__name__ + "-" + args.deploy_name if args.deploy_name else args.func.__name__
```

Comment:
Make this a multi-line if/else statement; its easier to read then squashing it on a single line.